### PR TITLE
Address clippy lints

### PR DESCRIPTION
**Pull request by benesch**
The first three commits are #119. I'll rebase if/when that merges.

### CODE REVIEW

File: docker/kafka_1.0.2/download-kafka.sh [hunk 1]
```diff
@@ -4,8 +4,10 @@ GREEN='\033[0;32m'
 RED='\033[0;31m'
 NC='\033[0m' # No Color
 
-mirror=$(curl --stderr /dev/null https://www.apache.org/dyn/closer.cgi\?as_json\=1 | jq -r '.preferred')
-url="${mirror}kafka/${KAFKA_VERSION}/kafka_${SCALA_VERSION}-${KAFKA_VERSION}.tgz"
+# We can't use a standard Apache mirror, since old versions of Kafka are
+# aggressively garbage collected. They live forever on archive.apache.org,
+# fortunately.
+url="https://archive.apache.org/dist/kafka/${KAFKA_VERSION}/kafka_${SCALA_VERSION}-${KAFKA_VERSION}.tgz"
```
Comment (flavray) [inline docker/kafka_1.0.2/download-kafka.sh]: good idea!

**Comment by flavray** [comment]
I've merged #119 (thanks again for the contribution!). Feel free to rebase this branch on top of `master` 🙂

**Comment by benesch** [comment]
Awesome! I've rebased this PR.

**Comment by flavray** [comment]
Thanks!

**Pull request merged**